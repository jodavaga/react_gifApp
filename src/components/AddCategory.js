import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('Hola mundo');

    // handle input change
    const handleChange = (e) => {
       setInputValue( e.target.value );
    }

    // handle ENTER, submit
    const handleSubmit = (e) => {
        e.preventDefault();

        if (!(inputValue.length > 2)) {
            return;
        }

        setCategories((cats) => [...cats, `${inputValue.charAt(0).toUpperCase()}${ inputValue.slice(1) }`]);
        setInputValue(''); 
    }

    return(
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                placeholder = "Add category"
                value={ inputValue }
                onChange={ handleChange }
                onClick={ (e) => e.target.setSelectionRange(0, e.target.value.length) }
            />
        </form>
    )
}

// Prop types
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}