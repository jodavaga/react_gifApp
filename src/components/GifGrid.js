import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import { GifItem } from './GifItem';

export const GifGrid = ({ category }) => {
    const apiKey = 'd50nWkDCfjhEfVgJIE9WYClyFqqYuGkZ';

    const [images, setImages] = useState([]);

    useEffect(() => {
        getGifs();
    }, []);

    const getGifs = async () => {

        const url = `https://api.giphy.com/v1/gifs/search?q=rick+and+morty&limit=10&api_key=${ apiKey }`;

        const resp = await fetch( url );
        const {data} = await resp.json();

        const gifs =  data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })

        console.log(gifs);
        // useState hook
        setImages( gifs );
    }

    return(
        <>
            <h2>{category}</h2>

            <div className="grid">
                {
                    images.map((data) => (
                        <GifItem 
                            key={ data.id }
                            {...data}
                        />
                    ))
                }
            </div>

        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

