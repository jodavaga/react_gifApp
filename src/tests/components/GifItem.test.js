import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GifItem } from '../../components/GifItem';

describe('<GifItem /> tests', () => {

    let wrapper = shallow(<GifItem />);

    beforeEach(() => {
        wrapper = shallow(<GifItem />);
    })

    test('Should render component', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('img').length).toBe(1);
    });
})