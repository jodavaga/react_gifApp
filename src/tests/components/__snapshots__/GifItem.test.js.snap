// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem /> tests Should render component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p />
</div>
`;
